### ISSUE TITLE
Shader link error using MaterialBuilder at runtime

### ISSUE THREAD

**Issue by rawnsley** [google/filament#5810]
Android app sample-material-builder fails at runtime with the following logcat error:

```
E/Filament: Link error in "Clear coat":
    "(null)"
```

The problem goes away if you swap from `MaterialBuilder.Optimization.NONE` to `MaterialBuilder.Optimization.PERFORMANCE` on [this line](https://github.com/google/filament/blob/release/android/samples/sample-material-builder/src/main/java/com/google/android/filament/material_builder/MainActivity.kt#L217).

I'm using the _release_ branch.

**Comment by romainguy** [comment]
Seems like a driver specific bug. On what device does this happen?

**Comment by romainguy** [comment]
Note btw that to use the `PERFORMANCE` mode you must not use `filamat-lite`

**Comment by romainguy** [comment]
@rawnsley Actually I think it might be an issue on our end, cf my previous comment.

@prideout We should prob degrade more gracefully when `PERFORMANCE` is used with `filamat-lite`.

**Comment by bejado** [comment]
I'm confused, @rawnsley is that link error generated when optimization is set to `MaterialBuilder.Optimization.NONE` or `MaterialBuilder.Optimization.PERFORMANCE`? I'm not sure this has anything to do with filamat-lite.

**Comment by rawnsley** [comment]
@bejado  I’m AFK at the moment, but I think the problem happens for NONE, i.e. the unchanged sample code with the full version of the material builder rather than the lite one. My memory is that it fails on both, but I might be confused.

**Comment by bejado** [comment]
Yes, I'm able to reproduce with the unchanged sample code (`MaterialBuilder.Optimization.NONE`). If I switch to `MaterialBuilder.Optimization.PERFORMANCE` the error is gone.

Unfortunately I think this is a driver bug. Here are the specs I can repro with:

```
FEngine resolved backend: OpenGL
QUALCOMM build                   : 85da404, I46ff5fc46f
Build Date                       : 11/30/20
OpenGL ES Shader Compiler Version: EV031.31.04.01
Local Branch                     : promo490_3_Google
Remote Branch                    :
Remote Branch                    :
Reconstruct Branch               :
Build Config                     : S P 10.0.4 AArch64
Driver Path                      : /vendor/lib64/egl/libGLESv2_adreno.so
```
